lib/paramedic.js

```diff
@@ -106,14 +106,21 @@ ParamedicRunner.prototype = {
         }).then(function(res) {
             if (self.testsPassed) {
                 logger.info("All tests have been passed.");
-                process.exit(0);
+                return 0;
             } else {
                 logger.error("There are tests failures.");
-                process.exit(1);
+                return 1;
             }
         }, function(err) {
             logger.error("Failed: " + err);
-            process.exit(2);
+            return 2;
+        }).then(function(exitCode){
+            if(self.config.isCleanUpAfterRun()) {
+                logger.info("cordova-paramedic: Deleting the application: " + self.tempFolder.name);
+                shell.cd("..");
```

Comment:
@sarangan12 please refactor the `cd` to be outside of `createTempProject`. This side effect makes the code harder to read and makes it easier to introduce the type of bug you described.